### PR TITLE
feat: Instant rewards smart contract

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Instant rewards smart contract

## Summary by CodeRabbit

- **New Features**
	- Introduced the `InstantRewards` contract, allowing users to claim rewards based on verified signatures.
	- Added secure claim functionality, ensuring each task can only be claimed once per user.
	- Implemented administrative functions for managing signer addresses and withdrawing funds.
	- Enhanced security features including ownership management, pausing functionality, and reentrancy protection.
- **Tests**
	- Added a comprehensive test suite for the `InstantRewards` contract, covering claim functionality, paused claims, and ownership transfer.

### CODE REVIEW

File: packages/zevm-app-contracts/test/instant-rewards/instant-rewards.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      value: amount,
+    });
+
+    const claimDataSigned = await getClaimDataSigned(signer, amount, sigExpiration, taskId, to);
+
+    const tx = instantRewards.claim(claimDataSigned);
+    await expect(tx).to.emit(instantRewards, "Claimed").withArgs(owner.address, taskId, amount);
+  });
+
+  it("Should revert if try to claim on behalf of somebody else", async () => {
```
Comment (owentar) [inline packages/zevm-app-contracts/test/instant-rewards/instant-rewards.ts:89]: I think we're missing one scenario to test: `Should revert if try to claim twice`?

File: packages/zevm-app-contracts/contracts/instant-rewards/InstantRewards.sol [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    struct Signature {
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    struct ClaimData {
+        address to;
+        Signature signature;
+        uint256 sigExpiration;
```
Comment (owentar) [inline packages/zevm-app-contracts/contracts/instant-rewards/InstantRewards.sol:20]: nit: not sure if this is something intended or not? missing the `n` on `sign`?

**Review by owentar — APPROVED** [review]
Left really minor comments

File: packages/zevm-app-contracts/test/instant-rewards/instant-rewards.ts (hunk 1 above)
Comment (andresaiello) [reply packages/zevm-app-contracts/test/instant-rewards/instant-rewards.ts:90]: added

File: packages/zevm-app-contracts/contracts/instant-rewards/InstantRewards.sol [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        if (signerAddress_ == address(0)) revert InvalidAddress();
+        signerAddress = signerAddress_;
+    }
+
+    function withdraw(address wallet, uint256 amount) external onlyOwner {
+        if (wallet == address(0)) revert InvalidAddress();
+        if (amount > address(this).balance) revert TransferFailed();
+        payable(wallet).transfer(amount);
+        emit Withdrawn(wallet, amount);
+    }
```
[the remark this answers is not in the document]
Comment (andresaiello) [reply packages/zevm-app-contracts/contracts/instant-rewards/InstantRewards.sol:96]: does not apply because we have a onlyOwner modifier

**Review by GMaiolo — APPROVED** [review]
utACK
LGTM

**Pull request merged**